File: ApiCallTest/app/src/main/java/com/example/raluc/apicalltest/viewModels/MainViewModel.java
package com.example.raluc.apicalltest.viewModels;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;
import android.content.Intent;
import android.util.Log;

import com.example.raluc.apicalltest.data.remote.RemoteServiceHelper;
import com.example.raluc.apicalltest.data.repository.RepositoryImpl;
import com.example.raluc.apicalltest.models.Album;
import com.example.raluc.apicalltest.ui.MainActivity;
import com.example.raluc.apicalltest.ui.SearchResults;

import java.util.List;


public class MainViewModel extends ViewModel {

    private static String TAG = "MainViewModel";



    public MainViewModel() {
        Log.d(TAG, "inMainViewModel");
    }

}

File: 18. Telegraph Challenge/app/src/main/java/com/example/raluc/telegraphchallenge/ui/MainViewModel.kt
package com.example.raluc.telegraphchallenge.ui

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.Transformations
import android.arch.lifecycle.ViewModel
import com.example.raluc.telegraphchallenge.adapters.PopularMoviesAdapter
import com.example.raluc.telegraphchallenge.data.repository.Repository

class MainViewModel(private val repository : Repository)  : ViewModel() {
    fun getPopularMovies():LiveData<PopularMoviesAdapter> {
        return Transformations.map(repository.getPopularMovies()){
            PopularMoviesAdapter(it.collection)
        }
    }
}